sdk/core/azure-core-amqp/src/models/amqp_value.cpp

```diff
@@ -0,0 +1,848 @@
+    }
+    return value;
+  }
+
+  AmqpValue::operator std::string() const
+  {
+    const char* value;
+    if (amqpvalue_get_string(m_value.get(), &value))
+    {
+      throw std::runtime_error("Could not retrieve value");
+    }
+    return value;
+  }
+
+  AmqpValue::operator Azure::Core::Uuid() const
+  {
+    uuid value;
+    if (amqpvalue_get_uuid(m_value.get(), &value))
+    {
+      throw std::runtime_error("Could not retrieve value");
+    }
+    std::array<uint8_t, 16> uuid;
+    memcpy(uuid.data(), value, 16);
+    return Azure::Core::Uuid::CreateFromArray(uuid);
+  }
+
+  bool AmqpValue::operator==(AmqpValue const& that) const
+  {
+    return amqpvalue_are_equal(m_value.get(), that.m_value.get());
+  }
+
+  bool AmqpValue::operator<(AmqpValue const& that) const
+  {
+    if (GetType() != that.GetType())
+    {
+      // If the types don't match, use the numeric type ordering to compare the types.
+      return GetType() < that.GetType();
+    }
+    switch (GetType())
+    {
+      case AmqpValueType::Null:
+        return false;
+      case AmqpValueType::Bool:
+        return static_cast<bool>(*this) < static_cast<bool>(that);
+      case AmqpValueType::Ubyte:
+        return static_cast<uint8_t>(*this) < static_cast<uint8_t>(that);
+      case AmqpValueType::Ushort:
+        return static_cast<uint16_t>(*this) < static_cast<uint16_t>(that);
+      case AmqpValueType::Uint:
+        return static_cast<uint32_t>(*this) < static_cast<uint32_t>(that);
+      case AmqpValueType::Ulong:
+        return static_cast<uint64_t>(*this) < static_cast<uint64_t>(that);
+      case AmqpValueType::Byte:
+        return static_cast<std::int8_t>(*this) < static_cast<std::int8_t>(that);
+      case AmqpValueType::Short:
+        return static_cast<int16_t>(*this) < static_cast<int16_t>(that);
+      case AmqpValueType::Int:
+        return static_cast<int32_t>(*this) < static_cast<int32_t>(that);
+      case AmqpValueType::Long:
+        return static_cast<int64_t>(*this) < static_cast<int64_t>(that);
+      case AmqpValueType::Float:
+        return static_cast<float>(*this) < static_cast<float>(that);
+      case AmqpValueType::Double:
+        return static_cast<double>(*this) < static_cast<double>(that);
+      case AmqpValueType::Char:
+        return static_cast<char32_t>(*this) < static_cast<char32_t>(that);
+      case AmqpValueType::String:
+        return static_cast<std::string>(*this) < static_cast<std::string>(that);
+      case AmqpValueType::Symbol:
+        return AsSymbol() < that.AsSymbol();
+
+      case AmqpValueType::Map:
+        return AsMap() < that.AsMap();
+      case AmqpValueType::Array:
+        return AsArray() < that.AsArray();
+      case AmqpValueType::Timestamp:
+        return AsTimestamp() < that.AsTimestamp();
+      case AmqpValueType::Uuid:
+        return static_cast<Azure::Core::Uuid>(*this).AsArray()
+            < static_cast<Azure::Core::Uuid>(that).AsArray();
```

Comment:
What does it mean for one map, array, or guid to be less than another?

Reply:
For std::map: https://en.cppreference.com/w/cpp/container/map/operator_cmp
For std::array: https://en.cppreference.com/w/cpp/container/array/operator_cmp

For Guid, the underlying array is compared.